src/copy.c

```diff
@@ -431,37 +436,119 @@ static inline void
 TSCopyMultiInsertBufferCleanup(TSCopyMultiInsertInfo *miinfo, TSCopyMultiInsertBuffer *buffer)
 {
 	int i;
-	ResultRelInfo *result_relation_info;
 
 	/* Ensure buffer was flushed */
 	Assert(buffer->nused == 0);
 
-	ChunkInsertState *cis = ts_chunk_dispatch_get_chunk_insert_state(miinfo->ccstate->dispatch,
-																	 buffer->point,
-																	 on_chunk_insert_state_changed,
-																	 buffer->bistate);
-	result_relation_info = cis->result_relation_info;
-	Assert(result_relation_info != NULL);
-
 	FreeBulkInsertState(buffer->bistate);
 
 	/* Since we only create slots on demand, just drop the non-null ones. */
 	for (i = 0; i < MAX_BUFFERED_TUPLES && buffer->slots[i] != NULL; i++)
 		ExecDropSingleTupleTableSlot(buffer->slots[i]);
 
-	table_finish_bulk_insert(result_relation_info->ri_RelationDesc, miinfo->ti_options);
-
 	pfree(buffer->point);
 	pfree(buffer);
 }
 
+#if PG13_LT
+/* list_sort comparator to sort TSCopyMultiInsertBuffer by usage */
+static int
+TSCmpBuffersByUsage(const void *a, const void *b)
+{
+	int b1 = ((const TSCopyMultiInsertBuffer *) lfirst(*(ListCell **) a))->nused;
+	int b2 = ((const TSCopyMultiInsertBuffer *) lfirst(*(ListCell **) b))->nused;
+
+	Assert(b1 >= 0);
+	Assert(b2 >= 0);
+
+	return (b1 > b2) ? 1 : (b1 == b2) ? 0 : -1;
+}
+#else
+/* list_sort comparator to sort TSCopyMultiInsertBuffer by usage */
+static int
+TSCmpBuffersByUsage(const ListCell *a, const ListCell *b)
+{
+	int b1 = ((const TSCopyMultiInsertBuffer *) lfirst(a))->nused;
+	int b2 = ((const TSCopyMultiInsertBuffer *) lfirst(b))->nused;
+
+	Assert(b1 >= 0);
+	Assert(b2 >= 0);
+
+	return (b1 > b2) ? 1 : (b1 == b2) ? 0 : -1;
+}
+#endif
```

Comment:
```suggestion
/* list_sort comparator to sort TSCopyMultiInsertBuffer by usage */
#if PG13_LT
static int
TSCmpBuffersByUsage(const void *a, const void *b)
{
	int b1 = ((const TSCopyMultiInsertBuffer *) lfirst(*(ListCell **) a))->nused;
	int b2 = ((const TSCopyMultiInsertBuffer *) lfirst(*(ListCell **) b))->nused;
#else
static int
TSCmpBuffersByUsage(const ListCell *a, const ListCell *b)
{
	int b1 = ((const TSCopyMultiInsertBuffer *) lfirst(a))->nused;
	int b2 = ((const TSCopyMultiInsertBuffer *) lfirst(b))->nused;
#endif

	Assert(b1 >= 0);
	Assert(b2 >= 0);

	return (b1 > b2) ? 1 : (b1 == b2) ? 0 : -1;
}
```
Nit. Just a simplification. It's up to you.

Reply:
I thought about writing it that way too. But I decided to write the more verbose version to improve readability and stay consistent with other existing compare functions like [this one](https://github.com/timescale/timescaledb/blob/40a6c4cf87edb3ca261cebcf4b2c6af11fcc6f0e/src/compat/compat.h#L560).